src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/AssemblyIdentityMustMatchTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests.Rules
+{
+    public class AssemblyIdentityMustMatchTests : SdkTest
+    {
+        public AssemblyIdentityMustMatchTests(ITestOutputHelper log) : base(log) { }
+
+        [Fact]
+        public static void AssemblyNamesDoNotMatch()
+        {
+            IAssemblySymbol left = CSharpCompilation.Create("AssemblyA").Assembly;
+            IAssemblySymbol right = CSharpCompilation.Create("AssemblyB").Assembly;
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            Assert.Single(differences);
+            Assert.Equal("CP0003 : left assembly name 'AssemblyA' does not match with right assembly name 'AssemblyB'.", differences.First().ToString());
```

Comment:
For these tests we are not asserting the messages because they are localized. Instead do a contains: `AssemblyA` and COntaines `AssemblyB'. 

You can look at the other tests, but there is a CompatDifferenceComparer that doesn't compare the message. In order to assert the other properties of the difference. 

Also, shouldn't this diagnostic ID be `CP0004`?

Reply:
CP0004 is thrown when mapper doesnot find the exact assembly and set one of the inputs as null

Reply:
A right.

Reply:
I dont think its worth it. I am expecting the mapper scenario to be more common and  we will hit that and version, culture and token are more interesting scenrios here.